### PR TITLE
Add tests for a Book functionality

### PR DESCRIPTION
**Pull request by 4Vitalii5**
Add test cases for Book and Category entities (repository, service, controller).
You should add at least one test case for each service and controller layer.
In general, you should have at least 50% lines test coverage for BookController, CategoryController, BookService, and CategoryService classes
Please add a screenshot with the test coverage result into the PR description
https://snipboard.io/uadnK8.jpg

### CODE REVIEW

File: src/main/java/mate/academy/dto/book/BookDto.java [hunk 1]
```diff
@@ -16,4 +16,16 @@ public class BookDto {
     private String description;
     private String coverImage;
     private List<Long> categoryIds;
+
+    public BookDto(Long id, String title, String author, String isbn, BigDecimal price,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/dto/book/BookDto.java]: Just use Lombok annotation instead

File: src/main/java/mate/academy/model/Book.java [hunk 2]
```diff
@@ -47,4 +49,15 @@ public class Book {
     )
     private Set<Category> categories = new HashSet<>();
 
+    public Book(Long id, String title, String author, String isbn, BigDecimal price,
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Book.java]: same here

File: src/main/java/mate/academy/model/Category.java [hunk 3]
```diff
@@ -26,4 +28,10 @@ public class Category {
     private String description;
     @Column(nullable = false)
     private boolean isDeleted = false;
+
+    public Category(Long id, String name, String description) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/model/Category.java]: same here

File: src/test/java/mate/academy/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+public class BookControllerTest {
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @DisplayName("Verify creation of new book")
+    @WithMockUser(roles = "ADMIN")
+    @Sql(scripts = {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/controller/BookControllerTest.java]: replace scripts under test class

File: src/test/java/mate/academy/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+@Sql(scripts = "classpath:database/clean-database.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_CLASS)
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/controller/BookControllerTest.java]: minor comment BEFORE_TEST_CLASS is default phase, so you can avoid it

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments, apply it for all tests

**Review by okuzan — APPROVED** [review]

**Pull request merged**